src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc

```diff
@@ -396,13 +417,17 @@ BtreeLBAManager::_alloc_extents(
 		   c.trans, addr, total_len, hint, state.last_end);
 	    if (alloc_info.extent) {
 	      ceph_assert(alloc_info.val.is_paddr());
+	      assert(!alloc_info.extent->has_laddr() ||
+		alloc_info.extent->get_laddr() == iter.get_key());
```

Comment:
How about this? :
```c++
if (alloc_info.extent->has_laddr()) {
  assert(alloc_info.key == alloc_info.extent->get_laddr());
  assert(alloc_info.key == iter.get_key());
} else {
  alloc_info.extent->set_laddr(iter.get_key());
}
```

Reply:
I moved this check out to `BtreeLBAMapping::remap_mappings()`, please take a look again, thanks:-)